tools/please_pex/pex_main.py

```diff
@@ -141,27 +143,65 @@ def get_code(self, fullname):
         return module.__loader__.get_code(fullname)
 
 
+def pex_basepath(temp=False):
+    if temp:
+        import tempfile
+        return tempfile.mkdtemp(dir=os.environ.get('TEMP_DIR'), prefix='pex_')
+    else:
+        return os.path.expanduser('~/.pex')
+
+def pex_uniquedir():
+    return f'pex-{PEX_STAMP}'
```

Comment:
don't wanna spoil your fun but can we avoid fstrings here please? would be nice to keep compatibility with older pythons unless we have a strong need.